### PR TITLE
Revert "ddl: fix create partition table error under NO_UNSIGNED_SUBTRACTION"

### PR DESCRIPTION
**Pull request by xiongjiwei**
Reverts pingcap/tidb#25435

### What problem does this PR solve?

Problem Summary:

the PR introduces a bug https://github.com/pingcap/tidb/issues/26765, it fixes a corner case but breaks a more common case.  I am gonna revert PR pingcap/tidb#25435 and document the bad case in the issues https://github.com/pingcap/tidb/issues/24880. see https://github.com/pingcap/tidb/issues/25437#issuecomment-866966979 for more details.

### Check List 

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
fix a bug that creates partition fail if `NO_UNSIGNED_SUBTRACTION` is set.
```

### CODE REVIEW

File: table/tables/partition_test.go [hunk 1]
```diff
@@ -447,41 +448,25 @@ func (ts *testSuite) TestCreatePartitionTableNotSupport(c *C) {
...
-	ErrMsg3 := "[ddl:1493]VALUES LESS THAN value must be strictly increasing for each partition"
-	tk.MustExec("SET @@sql_mode='';")
-	tk.MustGetErrMsg(`CREATE TABLE tu3 like tu;`, ErrMsg3)
+	tk.MustExec(`create table t2 (a bigint unsigned) partition by range (a) (
+  						  partition p1 values less than (0),
+  						  partition p2 values less than (1),
+  						  partition p3 values less than (18446744073709551614),
+  						  partition p4 values less than (18446744073709551615),
+  						  partition p5 values less than maxvalue);`)
+	tk.MustExec("insert into t2 values(10);")
```
Comment (mjonss) [inline table/tables/partition_test.go:463]: Please also add something like
```
defer tk.MustExec(`drop table t2`)
```

**Review by mjonss — APPROVED** [review]
/LGTM (but I'm not yet an approver). But I would suggest you to add an additional cleanup after creating the table in the test.

File: table/tables/partition_test.go (hunk 1 above)
Comment (xiongjiwei) [reply table/tables/partition_test.go:465]: done

**Review by bb7133 — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**